### ISSUE TITLE
Multiple Root Objects in Core Data Post/Put

### ISSUE THREAD

**Issue by matej** [RestKit/RestKit#1762]
I was following [this wiki entry](https://github.com/RestKit/RestKit/wiki/Object-mapping#handling-multiple-root-objects-in-core-data-postput) and experimenting with nilling out the `targetObject` attribute, since I have multiple root objects in my a POST / PATCH responses.

Surprisingly my research indicates that the wiki demo example would actually probably work fine even without  setting the targetObject to nil. The `authors` key holds an array, which would cause RKMapperOperation to perform a [`mapRepresentations:atKeyPath:usingMapping:`](https://github.com/RestKit/RestKit/blob/development/Code/ObjectMapping/RKMapperOperation.m?source=cc#L297) call for that key (instead of `mapRepresentation:atKeyPath:usingMapping:`). The `mapRepresentations` call ignores the `targetObject`, so those objects get mapped using the usual object lookup / creation approach. 

In summary, it seems like as long as all secondary root keys are holding collections, one can still have the targetObject set, which is really convenient when you post a locally created object without an identification attribute assigned to it. 

It might be worth noting this in the documentation, if this is in fact something that is so by design, and not just a lucky coincidence that can break at any time.
